controller/controller.go

```diff
@@ -71,13 +76,27 @@ func initExtensionUpdatesFromDynamoDB() {
 	// Update the extensions map
 	for _, item := range result.Items {
 		id := *item["ID"].S
-		AllExtensionsMap.Store(id, extension.Extension{
+
+		ext := extension.Extension{
 			ID:          id,
 			Blacklisted: *item["Disabled"].BOOL,
 			SHA256:      *item["SHA256"].S,
 			Title:       *item["Title"].S,
 			Version:     *item["Version"].S,
-		})
+		}
+
+		var patchInfoMap map[string]*extension.PatchInfo
+		if item["PatchList"] != nil {
+			err := dynamodbattribute.UnmarshalMap(item["PatchList"].M, &patchInfoMap)
+			if err != nil {
+				log.Printf("failed to parse PatchList %v\n", err)
+				sentry.CaptureException(err)
+			}
+			ext.PatchList = patchInfoMap
+		}
```

Comment:
This place could be written more carefully and precise:
- one map access instead of two;
- assignment of `PatchList` only for successful case (currently it would do so even for the error case, which would work, but it leaves a possibility for error in the future);
- clearer intent.

```suggestion
		if plist := item["PatchList"]; plist != nil {
			var pinfo map[string]*extension.PatchInfo
			if err := dynamodbattribute.UnmarshalMap(plist.M, &pinfo); err != nil {
				log.Printf("failed to parse PatchList %v\n", err)
				sentry.CaptureException(err)
			} else {
				ext.PatchList = pinfo
			}
		}
```